trieste/acquisition/rule.py

```diff
@@ -154,8 +160,19 @@ def acquire(
         :param state: Unused.
         :return: The single (or batch of) points to query, and `None`.
         """
+
         acquisition_function = self._builder.prepare_acquisition_function(datasets, models)
-        return self._optimizer(search_space, acquisition_function), None
+        points = self._optimizer(search_space, acquisition_function)
+
+        if isinstance(self._builder, GreedyAcquisitionFunctionBuilder) and self._num_query_points>1:
```

Comment:
If you intend to call `prepare_acquisition_function(datasets, models)` for the first point regardless of whether it's greedy or not, you may need to give pending_points a default value in the classes above. The alternative is to fully separate GreedyAcquisitionFunctionBuilder calls from the other ones.